src/main/java/org/example/bookstoreapp/service/ShoppingCartService.java

```diff
@@ -0,0 +1,19 @@
+package org.example.bookstoreapp.service;
+
+import org.example.bookstoreapp.dto.ShoppingCart.ShoppingCartDto;
+import org.example.bookstoreapp.dto.cartItem.CartItemDto;
+import org.example.bookstoreapp.dto.cartItem.CartItemRequestDto;
+import org.example.bookstoreapp.dto.cartItem.CartItemUpdateRequestDto;
+import org.example.bookstoreapp.model.ShoppingCart;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingCartService {
+
+    ShoppingCartDto getShoppingCartByUser();
+
+    CartItemDto addBooksToShoppingCart(CartItemRequestDto requestDto);
+
+    CartItemDto update(Long id, CartItemUpdateRequestDto requestDto);
+
+    void delete(Long id);
```

Comment:
You could add more informative names, for example:

```suggestion
    CartItemDto addItemToCart(CartItemRequestDto requestDto);

    CartItemDto updateItemQuantity(Long id, CartItemUpdateRequestDto requestDto);

    void deleteItemFromCart(Long id);
```